finality-aleph/src/sync/handler/mod.rs

```diff
@@ -435,7 +435,7 @@ where
         header: J::Header,
     ) -> Result<(), <Self as HandlerTypes>::Error> {
         if let Err(e) = self.forest.update_body(&header) {
-            if matches!(e, ForestError::TooNew) {
```

Comment:
question: in case of ParentNotImported dont we want to do `header.id().number()`? I dont remember how exactly missed_data works

Reply:
The abstraction is that it remembers the highest block it failed to import and refetches at most up to that. This was supposed to work with just stuff missed because of being too new, but this only adds the missing parent problem, treating it exactly the same – it also saves the number.